lib/mongodb.js

```diff
@@ -802,6 +821,49 @@ MongoDB.prototype.count = function count(model, where, options, callback) {
 };
 
 /**
+ * Replace properties for the model instance data
+ * @param {String} model The name of the model
+ * @param {*} id The instance id
+ * @param {Object} data The model data
+ * @param {Object} options The options object
+ * @param {Function} [cb] The callback function
+ */
+MongoDB.prototype.replaceAttributes = function replaceAttrs(model, id, data, options, cb) {
+  if (this.debug)
+    debug('replaceAttributes', model, id, data);
+    
+  var oid = this.coerceId(model, id);
+  this.replaceWithOptions(model, oid, data, {upsert: false}, cb);
+};
+
+/**
+ * Update a model instance with id
+ * @param {String} model The name of the model
+ * @param {Object} id The id of the model instance
+ * @param {Object} data The property/value pairs to be updated or inserted if {upsert: true} is passed as options
+ * @param {Object} options The options you want to pass for update, e.g, {upsert: true}
+ * @callback {Function} [cb] Callback function
+ */
+MongoDB.prototype.replaceWithOptions = function(model, id, data, options, cb) {
+  this.execute(model, 'update', {_id: id}, data, options, function(err, info) {
+    if (this.debug)
+      debug('updateWithOptions.callback', model, {_id: id}, data, err, info);
+    
+    if (err)
+      return cb && cb(err);
```

Comment:
Nitpick (feel free to ignore): when both `if` and the next line are short enough, we usually join them to a single line:

```js
if (err) return cb && cb(err);
```

The same comment applies to most of your tests.